pkg/security/probe/probe_linux.go

```diff
@@ -1379,6 +1418,10 @@ func (p *Probe) ApplyRuleSet(rs *rules.RuleSet) (*kfilters.ApplyRuleSetReport, e
 		return nil, fmt.Errorf("failed to flush discarders: %w", err)
 	}
 
+	if !p.Config.RuntimeSecurity.ActivityDumpEnabled {
```

Comment:
How often is activity dump not enabled? Is it mostly in testing? My understanding is that activity dumps/sec profiles/anomaly detection is a hot new thing that's going to be a core feature of CWS.

Also is the reason that we do NOT want to flush syscalls when activity dump is enabled AND rulesets are reloaded is because with activity dump enabled, we want to record everything that happens for anomaly detection?